module/VuFindTheme/src/VuFindTheme/View/Helper/ThemeResources.php

```diff
@@ -56,37 +56,23 @@ public function __construct(\VuFindTheme\ResourceContainer $container)
     }
 
     /**
-     * Set up header items based on contents of theme resource container.
+     * Set up items based on contents of theme resource container.
      *
-     * @return void
-     */
-    public function __invoke()
-    {
-        // Add various types of content to the header:
-        $this->addMetaTags();
-        $this->addLinks();
-        $this->addScripts();
-    }
-
-    /**
-     * Given a colon-delimited configuration string, break it apart, making sure
-     * that URLs in the first position are not inappropriately split.
+     * @param string $position Position for the items to be inserted
+     * ('header' or 'footer')
      *
-     * @param string $current Setting to parse
-     *
-     * @return array
+     * @return void
      */
-    protected function parseSetting($current)
+    public function __invoke($position)
     {
-        $parts = explode(':', $current);
-        // Special case: don't explode URLs:
-        if (($parts[0] === 'http' || $parts[0] === 'https')
-            && '//' === substr($parts[1], 0, 2)
-        ) {
-            $protocol = array_shift($parts);
-            $parts[0] = $protocol . ':' . $parts[0];
+        if (!isset($position) || $position == 'header') {
+            $this->addMetaTags();
+            $this->addLinks();
+            $this->addScripts('header');
+        }
+        if (!isset($position) || $position == 'footer') {
+            return $this->addScripts('footer');
```

Comment:
Is a `return` really needed here?

Reply:
Yes, due to the differences between Laminas HeadScript and InlineScript View Helpers.

HeadScript just adds the script, but InlineScript returns a JS snippet that needs to be added to the corresponding spot in the template:
- https://docs.laminas.dev/laminas-view/helpers/head-script/
- https://docs.laminas.dev/laminas-view/helpers/inline-script/

Reply:
Okay, that makes sense... but then the @return tag on the comment for this function needs to be adjusted to indicate that a string may be returned (and maybe we should return an empty string or null the rest of the time for consistency).